daily-coding/118/cody.py

```diff
@@ -0,0 +1,62 @@
+def squareSort(numbers):
+    i = 0
+    j = len(numbers) - 1
+
+    if len(numbers) == 0:
+        return numbers
+
+    # all numbers are positive
+    if numbers[i] >= 0:
+        for m in range(len(numbers)):
+            numbers[m] = numbers[m]**2
+        return numbers
+
+    # all numbers are negative
+    if numbers[j] < 0:
+        m = j
+        for n in range(len(numbers)//2):
```

Comment:
How does this one work? How do you know you only need to do root(n) iterations?

Reply:
It iterates for half of numbers. n starts at the beginning and goes up and m starts at the end and goes down. I'm reversing the list while squaring them.

Reply:
What does // do in this?

Reply:
I just realized that it wasn't squaring the middle number if the list has an odd number of elements, so I added a fix.